middleware/common/source/communication/ipc.cpp

```diff
@@ -426,26 +433,42 @@ namespace casual
                      return Handle{ std::move( socket), ipc};
                   }
                } // create 
+
+
+               void block_writes( const std::filesystem::path& path)
+               {
```

Comment:
I think disallow_writes would be a better name. I associate the words "block" and "blocking" more as the opposite to "nowait".

Other possible name are deny_writes and forbid_writes